run-build-functions.sh

```diff
@@ -105,34 +105,50 @@ install_deps() {
 }
 
 restore_node_modules() {
-  local workspace_output
-  local workspace_exit_code
   local installer=$1
-  # YARN_IGNORE_PATH will ignore the presence of a local yarn executable (i.e. yarn 2) and default
-  # to using the global one (which, for now, is always yarn 1.x). See https://yarnpkg.com/configuration/yarnrc#ignorePath
-  # we can actually use this command for npm workspaces as well
-  workspace_output="$(YARN_IGNORE_PATH=1 "$INTERNAL_YARN_PATH/yarn" workspaces --json info 2>/dev/null)"
-  workspace_exit_code=$?
-  if [ $workspace_exit_code -eq 0 ]
-  then
-    echo "$installer workspaces detected"
-    local package_locations
-    # Extract all the packages and respective locations. .data will be a JSON object like
-    # {
-    #   "my-package-1": {
-    #     "location": "packages/blog-1",
-    #     "workspaceDependencies": [],
-    #     "mismatchedWorkspaceDependencies": []
-    #   },
-    #   (...)
-    # }
-    # We need to cache all the node_module dirs, or we'll always be installing them on each run
-    mapfile -t package_locations <<< "$(echo "$workspace_output" | jq -r '.data | fromjson | to_entries | .[].value.location')"
-    restore_js_workspaces_cache "${package_locations[@]}"
-  else
-    echo "No $installer workspaces detected"
-    restore_cwd_cache node_modules "node modules"
-  fi
+
+	if has_feature_flag "$featureFlags" "build-image_use_new_package_manager_detection"; then
+      if [ "$NETLIFY_BUILD_DEBUG" ]; then
+        echo "BUILD_INFO:\n$buildinfo\n"
+      fi
```

Comment:
Happy to keep this debug log here too (as it's closer to where it's used). But if it isn't the case already (I think we already do it) we should also leverage Buildbot's syslog to log the build info object too (and probably leveraging the log formatted fields) as it's then easier to reason with in ou log aggregator.

Reply:
One of the reasons we should migrate to buildbot with this 😂

Reply:
will remove it as it's already inside builbot as you've said